src/gui/wizard/owncloudwizard.cpp

```diff
@@ -104,6 +112,33 @@ OwncloudWizard::OwncloudWizard(QWidget *parent)
 
     // allow Flow2 page to poll on window activation
     connect(this, &OwncloudWizard::onActivate, _flow2CredsPage, &Flow2AuthCredsPage::slotPollNow);
+
+    adjustWizardSize();
+}
+
+void OwncloudWizard::adjustWizardSize()
+{
+    QList<QSize> pageSizes;
+    const auto pIds = pageIds();
+    std::transform(pIds.begin(), pIds.end(), std::back_inserter(pageSizes), [this](int pageId) {
+        auto p = page(pageId);
+
+        p->adjustSize();
+
+        return p->sizeHint();
+    });
+
+    const auto maxPageSize = *std::max_element(pageSizes.begin(), pageSizes.end(), [](const QSize &size1, const QSize &size2) {
```

Comment:
For safety you might want to check or assert that `pageSizes` isn't empty before that call. Otherwise dereferencing the iterator might lead to issues.